src/Facebook/FacebookBatchRequest.php

```diff
@@ -85,17 +87,35 @@ public function add($request, $name = null)
             throw new \InvalidArgumentException('Argument for add() must be of type array or FacebookRequest.');
         }
 
+        if (!is_array($optionsOrName)) {
+            $this->add($request, ['name' => $optionsOrName]);
+
+            return $this;
+        }
+
+        if (null === $optionsOrName) {
```

Comment:
you must check for this one before `!is_array($optionsOrName)`. Currently you will never enter this case